src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -849,33 +851,57 @@ EnsureTableCanBeColocatedWith(Oid relationId, char replicationModel,
 
 
 /*
- * EnsureLocalTableEmptyIfNecessary only checks for emptiness if only an empty
- * relation can be distributed in given configuration.
- *
- * In some cases, it is possible and safe to send local data to shards while
- * distributing the table. In those cases, we can distribute non-empty local
- * tables. This function checks the distributionMethod and relation kind to
- * see whether we need to be ensure emptiness of local table. If we need to
- * be sure, this function calls EnsureLocalTableEmpty function to ensure
- * that local table does not contain any data.
+ * EnsureLocalTableEmptyIfNecessary errors out if the function should be empty
+ * according to ShouldLocalTableBeEmpty but it is not.
  */
 static void
 EnsureLocalTableEmptyIfNecessary(Oid relationId, char distributionMethod,
-								 bool viaDepracatedAPI)
+								 bool viaDeprecatedAPI)
 {
-	if (viaDepracatedAPI)
+	if (ShouldLocalTableBeEmpty(relationId, distributionMethod, viaDeprecatedAPI))
 	{
 		EnsureLocalTableEmpty(relationId);
 	}
+}
+
+
+/*
+ * ShouldLocalTableBeEmpty returns true if should the local table should be
+ * empty before creating a citus table.
+ * In some cases, it is possible and safe to send local data to shards while
+ * distributing the table. In those cases, we can distribute non-empty local
+ * tables. This function checks the distributionMethod and relation kind to
+ * see whether we need to be ensure emptiness of local table.
+ */
+static bool
+ShouldLocalTableBeEmpty(Oid relationId, char distributionMethod,
+						bool viaDeprecatedAPI)
+{
+	bool shouldLocalTableBeEmpty = false;
```

Comment:
Nit: Even though in this case it seems okay, in general the more local a variable is the shorter the name can be. Because if it is local then it is used in less places and close to each other so the context is already there. However for global ones, the name can be as expressive as it can be without being too long of course.